odl/test/solvers/functional/functional_test.py

```diff
@@ -510,52 +496,67 @@ def test_multiplication_with_vector(space):
     assert all_almost_equal(y_other_times_func(x), expected_result,
                             places=places)
 
+# Fixtures for test_functional_quadratic_perturb
+linear_term = simple_fixture('linear_term', [False, True])
+quadratic_term = simple_fixture('quadratic_term', [0.0, 2.13])
+
 
-def test_functional_linear_perturb(space):
-    """Test for the functional f(.) + <y, .>."""
+def test_functional_quadratic_perturb(space, linear_term, quadratic_term):
+    """Test for the functional f(.) + a | . |^2 + <y, .>."""
     # Less strict checking for single precision
     places = 3 if space.dtype == np.float32 else 5
 
-    # The translation; an element in the domain
-    linear_term = noise_element(space)
-
-    # Creating the functional ||x||_2^2 and add the linear perturbation
     orig_func = odl.solvers.L2NormSquared(space)
-    functional = odl.solvers.FunctionalLinearPerturb(orig_func, linear_term)
+
+    if linear_term:
+        linear_term_arg = None
+        linear_term = space.zero()
+    else:
+        linear_term_arg = linear_term = noise_element(space)
+
+    # Creating the functional ||x||_2^2 and add the quadratic perturbation
+    functional = odl.solvers.FunctionalQuadraticPerturb(
+        orig_func, quadratic_term=quadratic_term, linear_term=linear_term_arg)
 
     # Create an element in the space, in which to evaluate
     x = noise_element(space)
 
     # Test for evaluation of the functional
-    assert all_almost_equal(functional(x), x.norm()**2 + x.inner(linear_term),
+    assert all_almost_equal(functional(x),
+                            (orig_func(x) +
+                             quadratic_term * x.inner(x) +
+                             x.inner(linear_term)),
```

Comment:
> I'm honestly not so sure about these kinds of tests. What I mean by "these tests" is tests of the implementation of something rather than the intended functionality of something.

Well (imo) these dont test anything right now, but are only there as regression tests to give us a huge red flag if something gets changed so we actually notice. I don't really see any issue with that. They also make sure the code is run at least once which stops the worst of bugs.

With that said, you are certainly correct in that this could be done in a better manner, but that would require quite some effort here, something I probably wouldn't feel is warranted for a function that is as specialized as this once.

In other places, like `Gradient` which is called all the time, I certainly agree and a proper look at those tests is sorely needed.